python-squid/src/main/java/org/sonar/python/DocstringVisitor.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * SonarQube Python Plugin
+ * Copyright (C) 2011-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.python;
+
+import com.sonar.sslr.api.AstNode;
+import com.sonar.sslr.api.AstVisitor;
+import com.sonar.sslr.api.Grammar;
+import com.sonar.sslr.api.Token;
+import java.util.HashMap;
+import java.util.Map;
+import javax.annotation.Nullable;
+import org.sonar.python.api.PythonGrammar;
+import org.sonar.python.api.PythonTokenType;
+import org.sonar.squidbridge.SquidAstVisitor;
+
+/**
+ * Visitor that retrieves all docstrings from a Python file.
+ * <p>
+ * Reminder: a docstring is a string literal that occurs as the first statement
+ * in a module, function, class, or method definition.
+ */
+public class DocstringVisitor<G extends Grammar> extends SquidAstVisitor<G> implements AstVisitor {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) "AstVisitor" is implemented by a super class; there is no need to implement it here. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1939)